react/features/base/media/middleware.js

```diff
@@ -89,16 +115,65 @@ function _setRoom({ dispatch, getState }, next, action) {
         // We don't have startWithAudioMuted and startWithVideoMuted here:
         jwt: false
     };
-    const audioMuted
-        = roomIsValid
-            ? Boolean(
-                getPropertyValue(state, 'startWithAudioMuted', mutedSources))
-            : _AUDIO_INITIAL_MEDIA_STATE.muted;
-    const videoMuted
-        = roomIsValid
-            ? Boolean(
-                getPropertyValue(state, 'startWithVideoMuted', mutedSources))
-            : _VIDEO_INITIAL_MEDIA_STATE.muted;
+
+    if (typeof audioMuted === 'undefined') {
```

Comment:
I wonder if readability would be increased if the assignment of audioMuted, videoMuted, and facing Mode were all done in the same place, instead of previous versions set and then truthy check done later on. The downside would be duplicate checks for error.

Reply:
Originally I was thinking about that too, but then I thought that would be a 3 way conditional value (or if clause), as we need to set different values for:

(!room)
(room && !error)
(room && error)

Or do you see an easier way to do it that I may not?

Reply:
Maybe something like this for thought food? I could be missing a case or several though.
```
const isRejoinCase = error && roomisValid;

// audio mute
if (isRejoinCase && audio) {
    audioMuted = audio.muted;
} else if (roomisValid) {
    audioMuted = Boolean(...);
} else {
    audioMuted = _AUDIO_INITIAL_MEDIA_STATE.muted;
}
```

Reply:
But maybe that's already what you meant, re-reading what you said.  If you find the current version readable, I'm okay with it because you were considering readability. I'm also okay with the rest of the changes but will leave final reviewing to other interested parties.

Reply:
Based on your comment and Saul's below (and a FIXME note in the code) I refactored the entire logic, I think this is now clear enough. Thanks for pushing this!